testing/go/types_test.go

```diff
@@ -2125,12 +2141,80 @@ var typesTests = []ScriptTest{
 				},
 			},
 			{
-				Query:    `SELECT text 'text' || ' and unknown';`,
-				Expected: []sql.Row{{"text and unknown"}},
+				// Create a unique, secondary index on a TEXT column
+				Query:    "CREATE UNIQUE INDEX v1_unique ON t_text(v1);",
+				Expected: []sql.Row{},
 			},
 			{
-				Query:    `SELECT text 'this is a text string' = text 'this is a text string' AS true;`,
-				Expected: []sql.Row{{"t"}},
+				Query: "SELECT * FROM t_text WHERE v1 = 'World';",
+				Expected: []sql.Row{
+					{2, "World"},
+				},
+			},
+			{
+				// Test the new unique constraint on the TEXT column
+				Query:       "INSERT INTO t_text VALUES (5, 'World');",
+				ExpectedErr: "duplicate unique key given: [World]",
+			},
+			{
+				Query: "SELECT * FROM t_text_unique WHERE v2 = '!';",
+				Expected: []sql.Row{
+					{4, nil, "!"},
+				},
+			},
+			{
+				Query: "SELECT * FROM t_text_unique WHERE v2 >= '!' ORDER BY v2;",
+				Expected: []sql.Row{
+					{4, nil, "!"},
+					{1, "Hello", "Bonjour"},
+					{2, "World", "tout le monde"},
+				},
+			},
+			{
+				// Test ordering by TEXT column in a secondary index
+				Query: "SELECT * FROM t_text_unique ORDER BY v2;",
+				Expected: []sql.Row{
+					{3, "", ""},
+					{4, nil, "!"},
+					{1, "Hello", "Bonjour"},
+					{2, "World", "tout le monde"},
+				},
+			},
+			{
+				Query: "SELECT * FROM t_text_unique ORDER BY id;",
+				Expected: []sql.Row{
+					{1, "Hello", "Bonjour"},
+					{2, "World", "tout le monde"},
+					{3, "", ""},
+					{4, nil, "!"},
+				},
+			},
+			{
+				Query:       "INSERT INTO t_text_unique VALUES (5, 'Another', 'Bonjour');",
+				ExpectedErr: "duplicate unique key given: [Bonjour]",
+			},
+			{
+				// Test that TEXT types can't be used in primary keys yet – Dolt doesn't support prefix lengths in
+				// primary indexes currently.
+				Query:       "CREATE TABLE t_text_pk (id TEXT PRIMARY KEY, col1 int);",
+				ExpectedErr: "blob/text column 'id' used in key specification without a key length",
```

Comment:
I don't think we should be enforcing unimplemented behavior in tests. Rather, I'd prefer to have a skipped test with the correct intended behavior that we can unskip later.

Reply:
I'm happy to pull this out if you feel strongly about it, but the intent was that since we know `TEXT` columns won't work correctly in primary keys, we should test that we are preventing users from running into that. It's trying to protect against the situation where you can create a primary key with a `TEXT` column, but then it doesn't work correctly when you try to use it.

Reply:
I see. In that case, I'd say that we should make it very clear in a comment that this test isn't correct, and is only enforcing the current behavior. I know you've already merged the PR, so we can make the change in a separate PR, perhaps as an addition to your next PR.